azurerm/resource_arm_analysis_services_server.go

```diff
@@ -0,0 +1,449 @@
+	}
+
+	future, err := client.Update(ctx, resGroup, name, analysisServicesServer)
+	if err != nil {
+		return fmt.Errorf("Error creating Analysis Services Server %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for completion of Analysis Services Server %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	server, err := client.GetDetails(ctx, resGroup, name)
+	if err != nil {
+		return err
+	}
+
+	if server.ID == nil {
+		return fmt.Errorf("Cannot read ID of Analysis Services Server %q (Resource Group %q)", name, resGroup)
+	}
+
+	d.SetId(*server.ID)
+
+	return resourceArmAnalysisServicesServerRead(d, meta)
+}
+
+func resourceArmAnalysisServicesServerDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).analysisServicesServerClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resGroup := id.ResourceGroup
+	name := id.Path["servers"]
+
+	future, err := client.Delete(ctx, resGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error deleting Analysis Services Server %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for deletion of Analysis Services Server %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	return nil
+}
+
+func validateAnalysisServicesServerName(v interface{}, k string) (warnings []string, errors []error) {
+	value := v.(string)
+
+	if !regexp.MustCompile(`^[a-z][0-9a-z]{2,62}$`).Match([]byte(value)) {
+		errors = append(errors, fmt.Errorf("%q must begin with a letter, be lowercase alphanumeric, and be between 3 and 63 characters in length", k))
+	}
+
+	return warnings, errors
+}
+
+func validateQuerypoolConnectionMode() schema.SchemaValidateFunc {
+	connectionModes := make([]string, len(analysisservices.PossibleConnectionModeValues()))
+	for i, v := range analysisservices.PossibleConnectionModeValues() {
+		connectionModes[i] = string(v)
+	}
+
+	return validation.StringInSlice(connectionModes, true)
+}
+
+func expandAnalysisServicesServerSku(sku string) *analysisservices.ResourceSku {
+	return &analysisservices.ResourceSku{
+		Name: &sku,
+	}
+}
+
+func flattenAnalysisServicesServerSku(sku *analysisservices.ResourceSku) string {
+	return *sku.Name
+}
+
+func expandAnalysisServicesServerProperties(d *schema.ResourceData) *analysisservices.ServerProperties {
+	adminUsers := expandAnalysisServicesServerAdminUsers(d)
+
+	serverProperties := analysisservices.ServerProperties{AsAdministrators: adminUsers}
+
+	//if backupBlobContainerUri, ok := d.GetOk("backup_blob_container_uri"); ok {
+	//	serverProperties.BackupBlobContainerURI = backupBlobContainerUri.(*string)
+	//}
+
+	serverProperties.IPV4FirewallSettings = expandAnalysisServicesServerFirewallSettings(d)
```

Comment:
I don't think there is a need for this function

Reply:
i'm not sure if i understand correctly. IMHO this function does more than i want to have simply inline. Can you describe in detail?

Reply:
it doesn't make much difference either way - so I think this is fine as-is